modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -499,8 +499,38 @@ def merge(self, right, **kwargs):
 
             kwargs["sort"] = False
 
-            def map_func(left, right=right_pandas, kwargs=kwargs):  # pragma: no cover
-                return pandas.merge(left, right_pandas, **kwargs)
+            def keep_index_or_not(left, right):
```

Comment:
```suggestion
            def keep_index(left, right):
```
I would prefer this name.

Reply:
this would then conflict with the local variable that is also named `keep_index`. I'm okay with the current naming, however, if you're willing to change the function name anyway, I would vote for `whether_keep_index()`

Reply:
We could rename the var. `if whether_keep_index` looks a little strange.

Reply:
I like `whether_keep_index`, it's easier to rename the function name than the variable inside. @YarShev are you ok with that?

Reply:
`should_keep_index` is also fine for me.